README.md

```diff
@@ -247,11 +247,223 @@ Unlike with local kernels, Hydrogen does not kill remote kernels when it disconn
 
 **Note:** Unlike a notebook server, the jupyter kernel gateway by default disables listing already-running kernels. This means that once disconnected from a kernel, you will not be able to reconnect to it. You can set `c.KernelGatewayApp.list_kernels = True` in your kernel gateway configuration to change this behavior.
 
+### Example remote kernel gateway
+
+To create a remote kernel gateway you could
+
+- Install Jupyter Kernel Gateway:
+
+```bash
+pip install jupyter_kernel_gateway
+```
+
+- Run the kernel gateway with parameters specifying the IP address and port to listen to, e.g.:
+
+```bash
+jupyter kernelgateway --ip=0.0.0.0 --port=8888
+```
+
+**Note**: "`0.0.0.0`" means "listen to all IPs, including public IPs".
+
+## Docker execution via kernel gateways
+
+You can use the same technique to create a kernel gateway in a Docker container. That would allow you to develop from Atom but with all the dependencies, autocompletion, environment, etc. of a Docker container.
+
+But, due to the way that the kernel gateway creates sub-processes for each kernel, you have to use it in a special way, you can't run the `jupyter kernelgateway` directly in your `Dockerfile` `CMD` section. You need to call it with an init manager such as [tini](https://github.com/krallin/tini) or run it from an interactive console.
+
+Here's an example of how to setup a Docker execution environment with Hydrogen running the kernel gateway from an interactive console:
+
+### Example Docker kernel gateway
+
+#### Dockerfile
+
+- Create a `Dockerfile`
+- Install `jupyter_kernel_gateway` in your `Dockerfile`
+- Expose the gateway port, in this example it will be `8888`:
+
+```Dockerfile
+FROM python:2.7
+
+# Remove in production
```

Comment:
That is just a comment that could be omitted, but it might remind someone that those 2 lines are only for development.

The block (of 2 lines) in the `Dockerfile` below the comment are the ones to be removed or commented out in production. For development it would look like:

```Dockerfile
# Remove in production
RUN pip install jupyter_kernel_gateway
EXPOSE 8888
```

And for production it could be:

```Dockerfile
# Remove in production
# RUN pip install jupyter_kernel_gateway
# EXPOSE 8888
```

Although, as commented above, the user could test his "production" environment without commenting out those lines in his `Dockerfile`, just by running the container normally, without overriding the default command with the loop and without having to re-build the image.